dev/tools/gen_keycodes/data/keyboard_key.tmpl

```diff
@@ -0,0 +1,238 @@
+// found in the LICENSE file.
+
+// DO NOT EDIT -- DO NOT EDIT -- DO NOT EDIT
+// This file is generated by dev/tools/gen_keycodes/bin/gen_keycodes.dart and
+// should not be edited directly.
+//
+// Edit the template dev/tools/gen_keycodes/data/keyboard_key.tmpl instead.
+// See dev/tools/gen_keycodes/README.md for more information.
+
+import 'package:flutter/foundation.dart';
+
+/// A class with static values that describe the keys that are returned from
+/// [RawKeyEvent.logicalKey].
+///
+/// These represent *logical* keys, which are keys which are interpreted in the
+/// context of any modifiers, modes, or keyboard layouts which may be in effect.
+///
+/// This is contrast to [PhysicalKeyboardKey], which represents a physical key
+/// in a particular location on the keyboard, without regard for the modifier
+/// state, mode, or keyboard layout.
+///
+/// See also:
+///
+///  * [RawKeyEvent], the keyboard event object received by widgets that listen
+///    to keyboard events.
+///  * [RawKeyboardListener], a widget used to listen to and supply handlers for
+///    keyboard events.
+class LogicalKeyboardKey extends Diagnosticable {
+  /// Defines a KeyboardKey value and optional debug name.
+  ///
+  /// To save executable size, it is recommended that the [debugName] be null in
+  /// release mode. You can do this using the [kReleaseMode] constant:
+  ///
+  /// {@tool sample}
+  /// const LogicalKeyboardKey mySpecialKey = LogicalKeyboardKey(
+  ///   0x0010000000a,
+  ///   debugName: kReleaseMode ? null : 'Special Key',
+  /// );
+  /// {@end-tool}
+  const LogicalKeyboardKey(this.keyId, {this.debugName, this.keyLabel});
+
+  /// A unique code representing this key.
+  ///
+  /// This is an opaque identifier, and should not be unpacked to derive
+  /// information from it, as the representation of the code could change at any
+  /// time.
+  final int keyId;
+
+  /// The debug string to print for this keyboard key, which will be null in
+  /// release mode.
+  final String debugName;
+
+  /// The Unicode string representing the character produced by a [RawKeyEvent].
+  ///
+  /// This value is useful for describing or matching mnemonic keyboard
+  /// shortcuts.
+  ///
+  /// On most platforms this is a single code point, but it could contain any
+  /// Unicode string. The `keyLabel` differs from [RawKeyEvent.character]
+  /// because `keyLabel` only takes into account the key being pressed, not any
+  /// combining keys pressed before it, so, for example, an “o” that follows a
+  /// combining dieresis (“¨”, COMBINING DIAERESIS (U+0308)) would just return
+  /// “o” for [keyLabel], but would return “ö” for [RawKeyEvent.character].
+  ///
+  /// {@macro flutter.services.RawKeyEventData.keyLabel}
+  final String keyLabel;
+
+  @override
+  int get hashCode => keyId.hashCode;
+
+  @override
+  bool operator ==(dynamic other) {
+    if (other.runtimeType != runtimeType) {
+      return false;
+    }
```

Comment:
`final LogicalKeyboardKey typedOther = other;`
Otherwise other.keyId is a dcall

Reply:
Done.